aesm-client/src/imp/windows.rs

```diff
@@ -1,314 +1,119 @@
+            .field("init_quote", &(self.init_quote as *const c_void))
+            .field(
+                "get_quote",
+                &(self.get_quote as *const c_void),
+            )
+            .field(
+                "get_launch_token",
+                &(self.get_launch_token as *const c_void),
+            )
+            .finish()
+    }
 }
 
-impl Drop for AesmInterface {
-    fn drop(&mut self) {
+impl std::default::Default for Library {
+    fn default() -> Self {
         unsafe {
-            if let Some(release) = (*(*self.inner).vtbl).release {
-                release(self.inner);
+            let library = dl::Library::new(LIBRARY)
+                .expect("Couldn't find sgx_uae_service.dll");
+            let init_quote = *library.get::<SgxInitQuoteFn>(SYM_SGX_INIT_QUOTE)
+                .expect("failed to get dll api function sgx_init_quote");
+            let get_quote = *library.get::<SgxGetQuoteFn>(SYM_SGX_GET_QUOTE)
+                .expect("failed to get dll api function sgx_get_quote");
+            let get_launch_token = *library.get::<GetLaunchTokenFn>(SYM_GET_LAUNCH_TOKEN)
+                .expect("failed to get dll api function get_launch_token");
+
+            Library {
+                library,
+                init_quote,
+                get_quote,
+                get_launch_token,
             }
-            CoUninitialize();
         }
     }
 }
 
 impl AesmClient {
     pub fn new() -> Self {
-        Default::default()
+        AesmClient::default()
     }
 
     pub fn try_connect(&self) -> Result<()> {
-        self.create_instance().map(|_| ())
+        // TODO: use QueryServiceStatusEx
+        libloading::Library::new("sgx_uae_service.dll")?;
+        Ok(())
     }
 
-    fn create_instance(&self) -> Result<AesmInterface> {
-        let mut interface: *mut InnerAesmInterface = std::ptr::null_mut();
-        unsafe {
-            CoInitializeEx(
-                ptr::null_mut(),
-                COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE,
-            )
-            .into_io_error()
-            .map_err(|e| {
-                IoError::new(ErrorKind::Other, format!("Failed to initialize COM: {}", e))
-            })?;
-
-            CoCreateInstance(
-                &CLSID_AESMINTERFACE,
-                ptr::null_mut(),
-                CLSCTX_ALL,
-                &IID_IAESMINTERFACE,
-                &mut interface as *mut _ as *mut *mut c_void,
-            )
-            .into_io_error()
-            .map_err(|e| {
-                if let Some(release) = (*(*interface).vtbl).release {
-                    release(interface);
-                }
-                CoUninitialize();
-                IoError::new(
-                    ErrorKind::Other,
-                    format!("Fail to create Aesm Interface {}", e),
-                )
-            })?;
-        }
-        Ok(AesmInterface { inner: interface })
+    fn get_library(&self) -> Arc<Library> {
```

Comment:
I don't think this function is necessary. You can just write `&self.library` where-ever you use this.